docs/modules/data-structures/pages/managing-map-memory.adoc

```diff
@@ -120,40 +120,17 @@ memory size percentage for each Hazelcast instance.
 [[understanding-map-eviction]]
 ==== Understanding Size Attributes
 
-Hazelcast measures the size of maps based on partitions. For example, when you specify a size using
-the `PER_NODE` attribute for `max-size`, Hazelcast calculates the maximum size for every partition. Hazelcast uses the following equation to calculate the maximum size of a partition:
+Hazelcast measures the size of maps by partitions. For example, if you use the `PER_NODE` attribute for `max-size`, Hazelcast calculates the maximum number of map entries in each cluster member for every partition. Hazelcast uses the following equation to calculate the maximum size of a partition:
 
 ```
 partition-maximum-size = max-size * member-count / partition-count
 ```
 
-NOTE: If the `partition-maximum-size` is less than 1 in the equation above, it will be set to 1
-(otherwise, the partitions would be emptied immediately by eviction due to the exceedance of
-`max-size` being less than 1).
+NOTE: If the `partition-maximum-size` is less than 1, it will be set to 1. Otherwise, all partitions would be emptied immediately.
 
-The eviction process starts according to this calculated partition maximum size when you try
-to put an entry. When entry count in that partition exceeds partition maximum size, eviction
-starts on that partition.
+The eviction process starts when you try
+to add an entry to a map. If the entry count in the entry's partition exceeds the `partition-maximum-size`, eviction starts on that partition.
```

Comment:
Maybe something like:

Entries are evicted when you try to add an entry to a map only if the entry count in the entry's partition exceeds the `partition-maximum-size`.